### DISCUSSION TITLE
CI: Use of real environment instead of ubuntu cross-compile

### DISCUSSION THREAD

**Discussion by micaelmalta** [dogecoin/dogecoin#2334] [Q&A]
Hello everyone,

Today, we are doing some tricks to make the CI cross-compile Windows and MacOS. This is not an ideal solution as this might not build in real world.
 
Why don't we use real environments for Windows and MacOS aka https://github.com/actions/virtual-environments#available-environments ?

**Comment by AbcSxyZ** [comment]
Do you have an idea if there is a lot of user using Dogecoin core on Windows ? What type of user are they ? I'm wondering if there is a lot of windows devs who would work on/compile dogecoin core, and if there is a lot of regular holders (with windows potentially) who run a full node.

Actually, I'm wondering what end users Dogecoin core has. I suspect regular user use mainly exchange/spv/web wallets, I'm thinking a tiny percentage would run a full node. I suspect dogecoin core end users are more miners, pool, service provider, exchanges, etc. users with more technical tools/skills, but it's only supposition.

I'm regarding from the documentation perspective, wondering if we should care about windows, if we can not assume they have at least a WSL, vm, whatever and care only UNIX compliant OS. Could be relevant for dev, less for simple holder using the core on Windows.

**Comment by patricklodder** [comment]
The reason for using cross compilation on CI is because this is used for building binary releases using gitian. We can always add CI for compilation on platforms that are often used, but not used for releases, but are really a significant number of Dogecoin Core users self-compiling for macOS or windows? (or armhf, or aarch64?)
